spec/next_rails/bundle_report_spec.rb

```diff
@@ -64,12 +64,17 @@
     end
   end
 
-  describe ".compatibility" do
-    describe "output" do
-      it "returns ERB generated output" do
-        output = NextRails::BundleReport.erb_output({}, [], 7.0)
-        expect(output).to match "gems incompatible with Rails 7.0"
-      end
+  describe ".rails_compatibility" do
+    it "returns nil for invalid rails version" do
+      output = NextRails::BundleReport.rails_compatibility(rails_version: nil)
+      expect(output).to eq(nil)
```

Comment:
I think `output` is always `nil` even for valid rails versions, because the last line of the method is `puts ....` and puts prints to the console but returns `nil`

```rb
irb(main):001> output = puts "hola"
hola
=> nil
irb(main):002> output.inspect
=> "nil"
irb(main):003>
```

here you'll have to capture the `$stdout` and assert that it doesn't have any content https://stackoverflow.com/questions/14987362/how-can-i-capture-stdout-to-a-string#22777806

Reply:
in the previous test `output = ...` was working because it was testing the return value of the `erb_output` method, that doesn't use `puts` and returns the string, but here for the `..._compatibility` methods the last line is `puts`

Reply:
@arielj thanks for checking it, and you are right it was checking the `era_output` that is why it was different.